theano/sandbox/gpuarray/basic_ops.py

```diff
@@ -343,43 +382,93 @@ def c_code(self, node, name, inputs, outputs, sub):
                                      PyArray_NDIM(%(inp)s),
                                      (size_t *)PyArray_DIMS(%(inp)s),
                                      (ssize_t *)PyArray_STRIDES(%(inp)s),
-                                     pygpu_default_context(),
+                                     %(ctx)s,
                                      Py_None);
         if (%(out)s == NULL) {
             %(fail)s
         }
-        """ % {'name': name, 'inp': inputs[0],
+        """ % {'name': name, 'inp': inputs[0], 'ctx': sub['context'],
```

Comment:
I think we need to document how context are handled. I mean, in the Theano/doc, somewhere, have a section that describe all that is needed for that.